app/assets/src/components/views/sample_page/SamplePage.jsx

```diff
@@ -0,0 +1,328 @@
+    if (this.pipelineInProgress()) {
+      setTimeout(() => {
+        location.reload();
+      }, 300000);
+    }
+  }
+
+  generateGsnapFilterStatus = jobStats => {
+    const { jobStatistics, hostGenome } = this.props;
+    if (!jobStatistics || !hostGenome || hostGenome.name !== "Human") {
+      // only relevant for Human as of 5/21/2018
+      return null;
+    }
+    for (let stat of jobStats) {
+      if (stat["task"] === "gsnap_filter_out") {
+        return null;
+      }
+    }
+    return "gsnap filter on human/chimp genome was not run.";
+  };
+
+  refreshPage = overrides => {
+    const newParams = Object.assign({}, this.props.reportPageParams, overrides);
+    window.location =
+      location.protocol +
+      "//" +
+      location.host +
+      location.pathname +
+      "?" +
+      getURLParamString(newParams);
+  };
+
+  handleTabChange = tab => {
+    this.setState({ currentTab: tab });
+  };
+
+  toggleSampleDetailsSidebar = () => {
+    this.setState({
+      sampleDetailsSidebarVisible: !this.state.sampleDetailsSidebarVisible
+    });
+  };
+
+  handlePipelineVersionSelect = version => {
+    this.refreshPage({ pipeline_version: version });
+  };
+
+  handleMetadataUpdate = (key, newValue) => {
+    if (key === "name") {
+      this.setState({
+        sampleName: newValue
+      });
+    }
+  };
+
+  pipelineInProgress = () => {
+    if (this.props.pipelineRun && this.props.pipelineRun.finalized === 1) {
+      return false;
+    }
+    return true;
+  };
+
+  renderVersionDisplay = () => {
+    const { pipelineRun } = this.props;
+    if (pipelineRun && pipelineRun.version && pipelineRun.version.pipeline) {
+      const versionString = `v${pipelineRun.version.pipeline}`;
+      const alignmentDBString = pipelineRun.version.alignment_db
+        ? `, NT/NR: ${pipelineRun.version.alignment_db}`
+        : "";
+
+      return versionString + alignmentDBString;
+    }
+    return "";
+  };
+
+  renderPipelineWarnings = () => {
+    const warnings = [];
+
+    if (
+      !this.pipelineInProgress() &&
+      pipelineHasAssembly(this.props.pipelineRun) &&
+      this.props.pipelineRun.assembled !== 1
+    ) {
+      warnings.push("The reads did not assemble for this run.");
+    }
+
+    if (warnings.length > 0) {
+      const content = (
+        <div>
+          {warnings.map(warning => (
+            <div className={cs.warning} key={warning}>
+              {warning}
+            </div>
+          ))}
+        </div>
+      );
+      return (
+        <BasicPopup
+          trigger={
+            <i className={cx("fa fa-exclamation-circle", cs.warningIcon)} />
+          }
+          position="bottom left"
+          content={content}
+          wide="very"
+          horizontalOffset={15}
+        />
+      );
+    } else {
+      return null;
+    }
+  };
+
+  renderTab = () => {
+    if (this.state.currentTab === "Report") {
+      const waitingSpinner = (
+        <div className={cs.waitingContainer}>
```

Comment:
Should this be site-wide element?
I thought there was already one... but cannot find it, so wemight never have done it.

Reply:
We don't have a <Spinner> yet. Would be good to do at some point!